### PR TITLE
[react, wallets] Add imToken Wallet

### PR DESCRIPTION
**Pull request by makoshan**
## Problem solved
This PR adds support for the imToken wallet, including wallet IDs, connectors, locales.

## Changes made

- Added imToken wallet support
- Updated locales for imToken Wallet
- Implemented ImTokenConnector class for connecting to imToken Wallet
- Created ImTokenWallet class for imToken wallet integration
- The following files were skipped due to too many changes: packages/wallets/src/evm/wallets/imtoken-wallet.ts, - packages/react/src/wallet/wallets/imtoken/imtokenWallet.tsx

## How to test

- [ ] Automated tests: link to unit test file
- [ ] Manual tests: step by step instructions on how to test

## Contributor NFT

Paste in your wallet address below and we will airdrop you a special NFT when your pull request is merged. 

```Address: ```

---

## PR-Codex overview
This PR adds support for the imToken wallet across various files and modules.

### Detailed summary
- Added `imTokenWallet` to wallets and locales
- Created `ImTokenConnector` class
- Implemented `ImTokenWallet` class
- Updated package.json files
- Modified `useInstalledWallets` to check for imToken installation

> The following files were skipped due to too many changes: `packages/wallets/src/evm/wallets/imtoken.ts`, `packages/react/src/wallet/wallets/imtoken/imTokenWallet.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by jnsdls** [comment]
thanks for the contribution @makoshan - we'll review!

File: packages/react/src/evm/locales/types.ts [hunk 1]
```diff
@@ -183,6 +183,7 @@ export type ThirdwebLocale = {
       getStartedScreen: { instruction: string };
       scanScreen: { instruction: string };
     };
+    trustWallet: ExtensionAndQRScreensLocale;
```
Comment (allen-hsu) [inline packages/react/src/evm/locales/types.ts:186]: 這個應該要刪除

File: packages/react/src/wallet/hooks/useInstalledWallets.ts [hunk 2]
```diff
@@ -12,19 +12,22 @@ export function useInstalledWallets() {
...
   let isTrustWalletInstalled = false;
+  let isImTokenInstalled = false;
 
   const window_: Window | undefined = globalThis?.window;
   if (assertWindowEthereum(window_)) {
     isMetamaskInstalled = !!getInjectedMetamaskProvider();
     isCoinbaseWalletInstalled = !!getInjectedCoinbaseProvider();
     isZerionWalletInstalled = !!window_.ethereum?.isZerion;
     isTrustWalletInstalled = !!window_.ethereum?.isTrust;
+    isImTokenInstalled = !!window_.ethereum?.isImToken;
```
Comment (allen-hsu) [inline packages/react/src/wallet/hooks/useInstalledWallets.ts:23]: 沒有插件錢包..

File: packages/react/src/wallet/wallets/imtoken/imTokenWallet.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    connectUI: ConnectUI,
+    isInstalled: isInstalled,
+  };
+};
+
+function isInstalled() {
+  if (assertWindowEthereum(globalThis.window)) {
+    return !!globalThis.window.ethereum.isImToken;
+  }
+  return false;
```
Comment (allen-hsu) [inline packages/react/src/wallet/wallets/imtoken/imTokenWallet.tsx:52]: 直接return false

File: packages/react/src/wallet/wallets/imtoken/imTokenWallet.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
...jEuNTM1NyAzMS40OTkxIDE4LjY2NDEgMzMuMzQ5OCAxOS4wNjcgMzcuOTZDMTkuNDcyMiA0Mi42MTAxIDIzLjkyMjIgNDYuMjg0NSAyOS43NTU3IDQ1Ljc3MzRDMzYuMDUwOSA0NS4yMjIzIDQwLjQzMzcgNDAuMzExNCA0MC43NjM0IDMzLjQyMzRDNDAuNzYwMyAzMy4wNTg2IDQwLjgzNyAzMi42OTc1IDQwLjk4OCAzMi4zNjU1TDQwLjk5IDMyLjM1NzJDNDEuMDU3OCAzMi4yMTI4IDQxLjEzNzIgMzIuMDc0MiA0MS4yMjcyIDMxLjk0MjhDNDEuMzYxNiAzMS43NDA5IDQxLjUzMzggMzEuNTE4IDQxLjc1NjIgMzEuMjczOUM0MS43NTgzIDMxLjI2NzYgNDEuNzU4MyAzMS4yNjc2IDQxLjc2MjYgMzEuMjY3NkM0MS45MjQxIDMxLjA4NDcgNDIuMTE5NCAzMC44ODcgNDIuMzM5NyAzMC42NzQ1QzQ1LjA4OTcgMjguMDc1IDU0Ljk5MzEgMjEuOTQ0MiA2NC4zNTkyIDIzLjg4NTVDNjQuNTU3MyAyMy45MjggNjQuNzM2MSAyNC4wMzM0IDY0Ljg2OTMgMjQuMTg2MkM2NS4wMDI1IDI0LjMzOTEgNjUuMDgyNiAyNC41MzA4IDY1LjA5ODEgMjQuNzMzWiIgZmlsbD0id2hpdGUiLz4KPC9nPgo8L2c+CjxkZWZzPgo8bGluZWFyR3JhZGllbnQgaWQ9InBhaW50MF9saW5lYXJfMjIyXzQ3ODMiIHgxPSI3My41MDA5IiB5MT0iNS4zMTI1IiB4Mj0iMi44NzI0MSIgeTI9Ijc3LjY3NDciIGdyYWRpZW50VW5pdHM9InVzZXJTcGFjZU9uVXNlIj4KPHN0b3Agc3RvcC1jb2xvcj0iIzExQzREMSIvPgo8c3RvcCBvZmZzZXQ9IjEiIHN0b3AtY29sb3I9IiMwMDYyQUQiLz4KPC9saW5lYXJHcmFkaWVudD4KPGNsaXBQYXRoIGlkPSJjbGlwMF8yMjJfNDc4MyI+CjxyZWN0IHdpZHRoPSI4MCIgaGVpZ2h0PSI4MCIgZmlsbD0id2hpdGUiLz4KPC9jbGlwUGF0aD4KPC9kZWZzPgo8L3N2Zz4=", // 这里需要替换为imToken的图标数据
```
Comment (allen-hsu) [inline packages/react/src/wallet/wallets/imtoken/imTokenWallet.tsx:30]: 這個也不要寫中文

File: packages/react/src/wallet/wallets/imtoken/imTokenWallet.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  WalletConfig,
+  ConnectUIProps,
+} from "@thirdweb-dev/react-core";
+import { ImTokenWallet, assertWindowEthereum } from "@thirdweb-dev/wallets";
+import { handleWCSessionRequest } from "../handleWCSessionRequest";
+import { useTWLocale } from "../../../evm/providers/locale-provider";
+import { ExtensionOrWCConnectionUI } from "../_common/ExtensionORWCConnectionUI";
+
+const imTokenWalletUris = {
+  ios: "imtokenv2://",
```
Comment (allen-hsu) [inline packages/react/src/wallet/wallets/imtoken/imTokenWallet.tsx:12]: 這個要放連接

File: packages/react/src/wallet/wallets/imtoken/imTokenWallet.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  android: "imtokenv2://",
+  other: "imtokenv2://",
```
Comment (allen-hsu) [inline packages/react/src/wallet/wallets/imtoken/imTokenWallet.tsx:14]: other應該要拿掉或者去官方網站

File: .changeset/tall-moose-add-imToken.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
+---
+"@thirdweb-dev/wallets": patch
+"@thirdweb-dev/react": patch
+---
+
+[Wallets/React] Add imToken Wallet
```
Comment (jnsdls) [inline .changeset/tall-moose-add-imToken.md:6]:
```suggestion
Add imToken Wallet
```

**Review by jnsdls — APPROVED** [review]
LGTM

**Comment by jnsdls** [comment]
thanks! we'll get a release out today!

**Pull request merged**